lib/router/utils.js

```diff
@@ -11,6 +11,13 @@ if (!Array.isArray) {
 
 export var isArray = _isArray;
 
+/**
+  Determines if an object is Promise by checking if it is "thenable".
+**/
+export function isPromise(obj) {
+  return !!obj && (typeof obj === 'object' || typeof obj === 'function') && typeof obj.then === 'function';
```

Comment:
Can you tweak to:

```js
return ((typeof obj === 'object' && obj !== null) || typeof obj === 'function') && typeof obj.then === 'function';
```

We would prefer to only do the comparison to `null` (instead of `!!obj`) when it is an `object`.